client/services/FeedbackGeneratorService.js

```diff
@@ -53,8 +54,46 @@ tie.factory('FeedbackGeneratorService', [
       }
     };
 
+    var _getBuggyOutputTestFeedback = function (
+      failingTest, codeEvalResult) {
+      var hintIndex = 0;
+      var buggyMessages = failingTest.getMessages();
+      var feedback = FeedbackObjectFactory.create(false);
```

Comment:
Minor: consider moving this below, near where you do all the feedback-related stuff.